src/modules/TodoListArray.js

```diff
@@ -0,0 +1,53 @@
+class TodoListArray {
+  constructor() {
+    this.todos = JSON.parse(localStorage.getItem('todos')) || [];
+  }
+
+  addTodo(todo) {
+    this.todos.push(todo);
+    localStorage.setItem('todos', JSON.stringify(this.todos));
+  }
+
+  updateTodo(idx, newTodo) {
+    this.todos[idx - 1] = newTodo;
+    for (let i = 0; i < this.todos.length; i += 1) {
+      this.todos[i].idx = i + 1;
+    }
```

Comment:
Kindly, instead of using the traditional `for loop` make use of JavaScript built-in array method such as `forEach` method for simple syntax and better readability.